<!DOCTYPE html>
<html>
<head>
  <title>Annotation of lncRNA in farm animals - a community effort</title>
  <meta charset="utf-8">
  <meta name="description" content="Annotation of lncRNA in farm animals - a community effort">
  <meta name="author" content="Daniel Fischer (as part of the FAANG lncRNA group)">
  <meta name="generator" content="slidify" />
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta http-equiv="X-UA-Compatible" content="chrome=1">
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/default.css" media="all" >
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/phone.css" 
    media="only screen and (max-device-width: 480px)" >
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/slidify.css" >
  <link rel="stylesheet" href="libraries/highlighters/highlight.js/css/tomorrow.css" />
  <base target="_blank"> <!-- This amazingness opens all links in a new tab. -->  <link rel=stylesheet href="libraries/widgets/bootstrap/css/bootstrap.css"></link>
<link rel=stylesheet href="libraries/widgets/quiz/css/demo.css"></link>

  
  <!-- Grab CDN jQuery, fall back to local if offline -->
  <script src="http://ajax.aspnetcdn.com/ajax/jQuery/jquery-1.7.min.js"></script>
  <script>window.jQuery || document.write('<script src="libraries/widgets/quiz/js/jquery.js"><\/script>')</script> 
  <script data-main="libraries/frameworks/io2012/js/slides" 
    src="libraries/frameworks/io2012/js/require-1.0.8.min.js">
  </script>
  
  

</head>
<body style="opacity: 0">
  <slides class="layout-widescreen">
    
    <!-- LOGO SLIDE -->
        <slide class="title-slide segue nobackground">
  <hgroup class="auto-fadein">
    <h1>Annotation of lncRNA in farm animals - a community effort</h1>
    <h2>RNACentral Meeting, Hinxton, 6 November 2017</h2>
    <p>Daniel Fischer (as part of the FAANG lncRNA group)<br/>Natural Resources Institute Finland, Green Technology</p>
  </hgroup>
    <a href="https://github.com/fischuu/fischuu.github.io/presentations/RAndGoogleVisualization/zipball/gh-pages" class="example">
     Download
    </a>
  <article></article>  
  <footer class = 'license'>
    <a href='http://creativecommons.org/licenses/by-nc-sa/3.0/'>
    <img width = '80px' src = 'http://mirrors.creativecommons.org/presskit/buttons/88x31/png/by-nc-sa.png'>
    </a>
  </footer>
</slide>
    

    <!-- SLIDES -->
    <slide class="" id="slide-1" style="background:;">
  <hgroup>
    <h2>FAANG-COST Hackathon</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li><p>Jan Gorodkin organized a FAANG-COST hackathon in Copenhagen from 24.-25.11.2016</p></li>
<li><p>The purpose was to initiate a community effort to establish a ncRNA database for farm animals</p></li>
<li><p>I present here the current state of the lncRNA subpart of it</p></li>
<li><p>In total 11 persons are working at the moment in that subgroub and XXX in total for all different subgroups together</p></li>
<li><p>The focus is currently on 5 different species (cow, pig, horse, chicken, sheep)</p></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-2" style="background:;">
  <hgroup>
    <h2>Data collection and filtering</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li><p>RNA-seq datasets were collected from the European Nucleotide Archive (ENA)</p></li>
<li><p>In addition team members made own datasets available to the group</p></li>
<li><p>Any experiment that met the following criteria was included into the collection</p>

<ul>
<li>Illumina output only</li>
<li>Paired-end reads</li>
<li>At least 100bp</li>
<li>Strand specific libraries</li>
</ul></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-3" style="background:;">
  <hgroup>
    <h2>Quality control and Data curation</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li><p>All obtained raw files were quality controlled with </p>

<ul>
<li>FastQC</li>
<li>MultiQC</li>
<li>Tailored scripts (e.g. strandness validation)</li>
</ul></li>
<li><p>For all files, among others, the following meta-data were archived:</p>

<ul>
<li>Tissue type (Brenda Ontology terms)</li>
<li>Corresponding references</li>
<li>ENA accession number</li>
</ul></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-4" style="background:;">
  <hgroup>
    <h2>Pipelines established</h2>
  </hgroup>
  <article data-timings="">
    <p>We established two different pipelines to annotate lncRNA:</p>

<p>The &quot;HS&quot; pipeline:</p>

<pre><code>* HISAT2
* StringTie
* FEELnc
</code></pre>

<p>The &quot;SC&quot; pipeline:</p>

<pre><code>* STAR
* Cufflinks
* FEELnc
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-5" style="background:;">
  <hgroup>
    <h2>Used Genome Versions and available samples</h2>
  </hgroup>
  <article data-timings="">
    <table><thead>
<tr>
<th>Species</th>
<th align="center">Genome</th>
<th align="right">Avail. Samples from ENA</th>
</tr>
</thead><tbody>
<tr>
<td>Chicken</td>
<td align="center">Gallus_gallus-5.0</td>
<td align="right">171</td>
</tr>
<tr>
<td>Cow</td>
<td align="center">UMD_3.1.1</td>
<td align="right">447</td>
</tr>
<tr>
<td>Horse</td>
<td align="center">EquCab2.0</td>
<td align="right">62</td>
</tr>
<tr>
<td>Pig</td>
<td align="center">Sscrofa11.1</td>
<td align="right">149</td>
</tr>
<tr>
<td>Sheep</td>
<td align="center">Oar_v4.0</td>
<td align="right">195</td>
</tr>
</tbody></table>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-6" style="background:;">
  <hgroup>
    <h2>Collected tissues per samples</h2>
  </hgroup>
  <article data-timings="">
    <div style="text-align:center">
<img src="./2017.09.14-data_collected_tissue_21_07_2017_RNAonly.jpg" alt="Drawing" style="width: 700px;"/>
</div>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-7" style="background:;">
  <hgroup>
    <h2>Pipeline comparison</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li><p>Sarah Djebali performed an extensive comparison on the two different pipelines for cow and chicken</p></li>
<li><p>For that, we ran both pipelines with and without gene annotations</p></li>
<li><p>Tissues were considered individually and also merged (1,2 and all)</p></li>
<li><p>Total number of transcript sets to evaluate:</p>

<ul>
<li>1 gene annotation + 24 predictions for chicken</li>
<li>1 gene annotation + 32 predictions for cattle</li>
</ul></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-8" style="background:;">
  <hgroup>
    <h2>Summary of the comparison</h2>
  </hgroup>
  <article data-timings="">
    <div style="text-align:center">
<img src="./compSummary.png" alt="Drawing" style="width: 800px;"/>
</div>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-9" style="background:;">
  <hgroup>
    <h2>Impacts of the comparisons</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li><p>Stringtie has improved a lot between version 1.3.2 and 1.3.3</p></li>
<li><p>Many less unstranded transcripts</p></li>
<li><p>Length of exons of different types better</p></li>
<li><p>Less impacted by not having an annotation</p></li>
<li><p>More similar results on both species</p></li>
<li><p>Stringtie becomes a tool that can be envisionned for FAANG,
although its results are more dependent on the annotation
quality and this one can be very variable for livestock species</p></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-10" style="background:;">
  <hgroup>
    <h2>FEELnc performance for our data</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li><p>Each pipeline produced one gtf as input for FEELnc</p></li>
<li><p>The numer of reads to do lncRNA annotation is also crucial and should be at least 70 Mio reads</p></li>
<li><p>Most of the available datasets had less</p></li>
<li><p>Consequently, we chose 30 Mio as threshold for a high-coverage set</p></li>
<li><p>That means: 4 different input gtfs per species.</p></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-11" style="background:;">
  <hgroup>
    <h2>FEELnc results I</h2>
  </hgroup>
  <article data-timings="">
    <table><thead>
<tr>
<th>Species</th>
<th align="center">Input</th>
<th align="right"># of lncRNA</th>
</tr>
</thead><tbody>
<tr>
<td>Chicken</td>
<td align="center">HS</td>
<td align="right">31.208</td>
</tr>
<tr>
<td></td>
<td align="center">HS30</td>
<td align="right">10.225</td>
</tr>
<tr>
<td></td>
<td align="center">SC</td>
<td align="right">46.288</td>
</tr>
<tr>
<td></td>
<td align="center">SC30</td>
<td align="right">29.310</td>
</tr>
<tr>
<td>Cow</td>
<td align="center">HS</td>
<td align="right">26.231</td>
</tr>
<tr>
<td></td>
<td align="center">HS30</td>
<td align="right">8.219</td>
</tr>
<tr>
<td></td>
<td align="center">SC</td>
<td align="right">141.036</td>
</tr>
<tr>
<td></td>
<td align="center">SC30</td>
<td align="right">41.980</td>
</tr>
</tbody></table>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-12" style="background:;">
  <hgroup>
    <h2>FEELnc results II</h2>
  </hgroup>
  <article data-timings="">
    <table><thead>
<tr>
<th>Species</th>
<th align="center">Input</th>
<th align="right"># of lncRNA</th>
</tr>
</thead><tbody>
<tr>
<td>Horse</td>
<td align="center">HS</td>
<td align="right">12.848</td>
</tr>
<tr>
<td></td>
<td align="center">HS30</td>
<td align="right">9.159</td>
</tr>
<tr>
<td></td>
<td align="center">SC</td>
<td align="right">77.668</td>
</tr>
<tr>
<td></td>
<td align="center">SC30</td>
<td align="right">53.932</td>
</tr>
<tr>
<td>Sheep</td>
<td align="center">HS</td>
<td align="right">27.713</td>
</tr>
<tr>
<td></td>
<td align="center">HS30</td>
<td align="right">6.270</td>
</tr>
<tr>
<td></td>
<td align="center">SC</td>
<td align="right">49.724</td>
</tr>
<tr>
<td></td>
<td align="center">SC30</td>
<td align="right">31.172</td>
</tr>
</tbody></table>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-13" style="background:;">
  <hgroup>
    <h2>Upcoming tasks</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li><p>Between tissue comparison for lncRNA expressions</p></li>
<li><p>Between species comparison for lncRNA expressions</p></li>
<li><p>Trace the reasons for differences in pipelines</p></li>
<li><p>If possible, function interpretation of the lncRNA</p></li>
<li><p>Agree on publication strategy</p></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-14" style="background:;">
  <hgroup>
    <h2>Vision for the FAANG non-coding and structured RNA (ncRNA) database</h2>
  </hgroup>
  <article data-timings="">
    <div style="font-size=20pt;">
<ul>
<li> Compilation of small, long and structured RNA

<li> For multiple animal genomes
  <ul>
   <li> Syntenic ncRNAs
   <li> Tissue expression
   <li> QTL
   <li> Conservation of features
  </ul>
<li> Integration with ENSEMBL will feed into external resources e.g.:
  <ul>
   <li> The tissues database[*]
   <li> RAIN for RNA and protein interactions/associations [#]
  </ul>
</ul>

[*] https://tissues.jensenlab.org

[#] https://rth.dk/resources/rain

</div>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-15" style="background:;">
  <hgroup>
    <h2>Acknowledgement</h2>
  </hgroup>
  <article data-timings="">
    <p>The FAANG lncRNA group consists of the following members</p>

<table><thead>
<tr>
<th>lncRNA group</th>
<th>ncDB efforts</th>
</tr>
</thead><tbody>
<tr>
<td>Daniel Fischer</td>
<td>Jan Gorodkin</td>
</tr>
<tr>
<td>Frank Panitz</td>
<td>Christian Anton</td>
</tr>
<tr>
<td>Konstantinos Billis</td>
<td>Andreia Maral</td>
</tr>
<tr>
<td>Thomas Derrien</td>
<td></td>
</tr>
<tr>
<td>Lel Eory</td>
<td>Genomic group</td>
</tr>
<tr>
<td>Alicja Pacholewska</td>
<td>sncRNA group</td>
</tr>
<tr>
<td>Nadezhda Doncheva</td>
<td></td>
</tr>
<tr>
<td>Magda Mielczarek</td>
<td></td>
</tr>
<tr>
<td>Sarah Djebali</td>
<td></td>
</tr>
<tr>
<td>Christa Kühn</td>
<td></td>
</tr>
<tr>
<td>Oana Palasca</td>
<td></td>
</tr>
</tbody></table>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-16" style="background:;">
  <hgroup>
    <h2>Appendix: Chicken summary - individual</h2>
  </hgroup>
  <article data-timings="">
    <div style="text-align:center">
<img src="./chickenSummary_ind.png" alt="Drawing" style="width: 700px;"/>
</div>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-17" style="background:;">
  <hgroup>
    <h2>Appendix: Chicken summary - merged</h2>
  </hgroup>
  <article data-timings="">
    <div style="text-align:center">
<img src="./chickenSummary_merged.png" alt="Drawing" style="width: 700px;"/>
</div>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-18" style="background:;">
  <hgroup>
    <h2>Appendix: Cow summary - individual</h2>
  </hgroup>
  <article data-timings="">
    <div style="text-align:center">
<img src="./cowSummary_ind.png" alt="Drawing" style="width: 700px;"/>
</div>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-19" style="background:;">
  <hgroup>
    <h2>Appendix: Cow summary - merged</h2>
  </hgroup>
  <article data-timings="">
    <div style="text-align:center">
<img src="./cowSummary_merged.png" alt="Drawing" style="width: 700px;"/>
</div>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-20" style="background:;">
  <article data-timings="">
    
  </article>
  <!-- Presenter Notes -->
</slide>

    <slide class="backdrop"></slide>
  </slides>
  <div class="pagination pagination-small" id='io2012-ptoc' style="display:none;">
    <ul>
      <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=1 title='FAANG-COST Hackathon'>
         1
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=2 title='Data collection and filtering'>
         2
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=3 title='Quality control and Data curation'>
         3
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=4 title='Pipelines established'>
         4
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=5 title='Used Genome Versions and available samples'>
         5
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=6 title='Collected tissues per samples'>
         6
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=7 title='Pipeline comparison'>
         7
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=8 title='Summary of the comparison'>
         8
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=9 title='Impacts of the comparisons'>
         9
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=10 title='FEELnc performance for our data'>
         10
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=11 title='FEELnc results I'>
         11
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=12 title='FEELnc results II'>
         12
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=13 title='Upcoming tasks'>
         13
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=14 title='Vision for the FAANG non-coding and structured RNA (ncRNA) database'>
         14
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=15 title='Acknowledgement'>
         15
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=16 title='Appendix: Chicken summary - individual'>
         16
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=17 title='Appendix: Chicken summary - merged'>
         17
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=18 title='Appendix: Cow summary - individual'>
         18
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=19 title='Appendix: Cow summary - merged'>
         19
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=20 title='NA'>
         20
      </a>
    </li>
  </ul>
  </div>  <!--[if IE]>
    <script 
      src="http://ajax.googleapis.com/ajax/libs/chrome-frame/1/CFInstall.min.js">  
    </script>
    <script>CFInstall.check({mode: 'overlay'});</script>
  <![endif]-->
</body>
  <!-- Load Javascripts for Widgets -->
  <script src="libraries/widgets/bootstrap/js/bootstrap.min.js"></script>
<script src="libraries/widgets/bootstrap/js/bootbox.min.js"></script>
<script src="libraries/widgets/quiz/js/jquery.quiz.js"></script>
<script src="libraries/widgets/quiz/js/mustache.min.js"></script>
<script src="libraries/widgets/quiz/js/quiz-app.js"></script>

  <!-- MathJax: Fall back to local if CDN offline but local image fonts are not supported (saves >100MB) -->
  <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        inlineMath: [['$','$'], ['\\(','\\)']],
        processEscapes: true
      }
    });
  </script>
  <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/2.0-latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  <!-- <script src="https://c328740.ssl.cf1.rackcdn.com/mathjax/2.0-latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  </script> -->
  <script>window.MathJax || document.write('<script type="text/x-mathjax-config">MathJax.Hub.Config({"HTML-CSS":{imageFont:null}});<\/script><script src="libraries/widgets/mathjax/MathJax.js?config=TeX-AMS-MML_HTMLorMML"><\/script>')
</script>
<script>  
  $(function (){ 
    $("#example").popover(); 
    $("[rel='tooltip']").tooltip(); 
  });  
  </script>  
  
  <!-- LOAD HIGHLIGHTER JS FILES -->
  <script src="libraries/highlighters/highlight.js/highlight.pack.js"></script>
  <script>hljs.initHighlightingOnLoad();</script>
  <!-- DONE LOADING HIGHLIGHTER JS FILES -->
   
  </html>